File: apps/Lodgr.Api/Models/Room.cs
using System.ComponentModel.DataAnnotations;
using System.ComponentModel.DataAnnotations.Schema;

namespace Lodgr.Api.Models;

[Table("rooms")]
public class Room
{
    [Key]
    [Column("room_id")]
    public long RoomId { get; set; }

    [Column("building_id")]
    public long BuildingId { get; set; }

    [Required]
    [StringLength(20)]
    [Column("room_number")]
    public string RoomNumber { get; set; } = string.Empty;

    [Column("floor")]
    public int? Floor { get; set; }

    [Column("area", TypeName = "numeric(8,2)")]
    public decimal? Area { get; set; }

    [Column("base_price", TypeName = "numeric(12,2)")]
    public decimal BasePrice { get; set; }

    [Column("room_type")]
    public RoomType? RoomType { get; set; }

    [Range(1, int.MaxValue)]
    [Column("max_occupants")]
    public int MaxOccupants { get; set; } = 1;

    [Column("operational_status")]
    public OperationalStatus OperationalStatus { get; set; } = OperationalStatus.ACTIVE;

    [Column("description")]
    public string? Description { get; set; }

    [Column("created_at")]
    public DateTime CreatedAt { get; set; }

    [Column("updated_at")]
    public DateTime UpdatedAt { get; set; }

    [Column("is_deleted")]
    public bool IsDeleted { get; set; }
}

public enum RoomType
{
    SINGLE,
    DOUBLE,
    STUDIO,
    FAMILY,
}

public enum OperationalStatus
{
    ACTIVE,
    MAINTENANCE,
    BLOCKED,
}

File: apps/api/src/main/java/io/github/hoanghonghuy/lodgr/entity/Room.java
package io.github.hoanghonghuy.lodgr.entity;

import io.github.hoanghonghuy.lodgr.enums.OperationalStatus;
import io.github.hoanghonghuy.lodgr.enums.RoomType;
import jakarta.persistence.Column;
import jakarta.persistence.Entity;
import jakarta.persistence.EnumType;
import jakarta.persistence.Enumerated;
import jakarta.persistence.FetchType;
import jakarta.persistence.GeneratedValue;
import jakarta.persistence.GenerationType;
import jakarta.persistence.Id;
import jakarta.persistence.JoinColumn;
import jakarta.persistence.ManyToOne;
import jakarta.persistence.Table;
import lombok.Getter;
import lombok.Setter;

import java.math.BigDecimal;
import java.time.LocalDateTime;

@Entity
@Table(name = "rooms")
@Getter
@Setter
public class Room {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    @Column(name = "room_id")
    private Long roomId;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "building_id", nullable = false)
    private Building building;

    @Column(name = "room_number", nullable = false, length = 20)
    private String roomNumber;

    @Column(name = "floor")
    private Integer floor;

    @Column(name = "area", precision = 8, scale = 2)
    private BigDecimal area;

    @Column(name = "base_price", nullable = false, precision = 12, scale = 2)
    private BigDecimal basePrice;

    @Enumerated(EnumType.STRING)
    @Column(name = "room_type")
    private RoomType roomType;

    @Column(name = "max_occupants", nullable = false)
    private Integer maxOccupants;

    @Enumerated(EnumType.STRING)
    @Column(name = "operational_status", nullable = false)
    private OperationalStatus operationalStatus;

    @Column(name = "description")
    private String description;

    @Column(name = "created_at", nullable = false)
    private LocalDateTime createdAt;

    @Column(name = "updated_at", nullable = false)
    private LocalDateTime updatedAt;

    @Column(name = "is_deleted", nullable = false)
    private Boolean isDeleted;
}
